tasks/facts-ansible10.yml

```diff
@@ -0,0 +1,3 @@
+---
+- name: Gather Ansible Facts
+  ansible.builtin.setup:
```

Comment:
What do we need this for?

Reply:
This task fills the `ansible_facts` dict: https://github.com/DataDog/ansible-datadog/pull/304

For some reason, the unprefixed task failed for @KSerrania when he ran it on Linux if `ansible.windows` was installed (🤷), but the prefixed task doesn't work in old Ansible versions... so I put each on a separate files and we run one or the other conditionally.

Reply:
I see, that's weird but if that's the way it works this seems like the best we can do yeah

Reply:
Could you add a comment explaining this in the code? This is quite unintuitive and I see myself wondering in a month why we added this.